CDCgov/prime-reportstream
# 5754 Generate HL7 ORC segment from FHIR

This PR generates the ORC segment from FHIR data. Note the ORC segment is optional, and it is generated only when needed.

Test Steps:
1. Check out the branch for this PR.
2. Run the following command to convert a FHIR bundle to HL7.  
`./prime fhirdata --input-file "src/testIntegration/resources/datatests/HL7_to_FHIR/sample_co_1_20220518-0001.fhir" -s metadata/hl7_mapping/ORU_R01/ORU_R01-base.yml`
3. Verify an HL7 message is generated from the conversion and includes the ORC segments.
4. Look at the source HL7 used at `src/testIntegration/resources/datatests/HL7_to_FHIR/sample_co_1_20220518-0001.hl7` and verify the data for the ORC segments are identical.  There are exceptions:
   a. ISO is not added when there is no universal ID specified
   b. A person's suffix is added in the suffix field instead of the professional suffix or degree field per FHIR's spec location.
   c. Date times have added precision, but there is an issue open for that

## Changes
- Added config for the FHIR to HL7 conversion of the ORC segment
- Modified values for address type

## Checklist

### Testing

### Process

## Linked Issues
- #5754

## To Be Done

## Specific Security-related subjects a reviewer should pay specific attention to
